projects/plots/plots/vizapp/infer/analyzer.py

```diff
@@ -0,0 +1,213 @@
+            kernel_length,
+            sample_rate,
+            inference_sampling_rate,
+            batch_size,
+            fduration,
+            fftlength=fftlength,
+            highpass=highpass,
+            return_whitened=True,
+        ).to(device)
+
+        self.snapshotter = BackgroundSnapshotter(
+            psd_length=psd_length,
+            kernel_length=kernel_length,
+            fduration=fduration,
+            sample_rate=sample_rate,
+            inference_sampling_rate=inference_sampling_rate,
+        ).to(device)
+
+        self.response_set = response_set
+        self.strain_dir = strain_dir
+        self.ifos = ifos
+        self.padding = padding
+        self.sample_rate = sample_rate
+        self.fduration = fduration
+        self.psd_length = psd_length
+        self.kernel_length = kernel_length
+        self.inference_sampling_rate = inference_sampling_rate
+        self.integration_length = integration_length
+        self.batch_size = batch_size
+        self.device = device
+
+    @property
+    def waveform_class(self):
+        return waveform_class_factory(
+            self.ifos, InterferometerResponseSet, "IfoWaveformSet"
+        )
+
+    @property
+    def kernel_size(self):
+        return int(self.kernel_length * self.sample_rate)
+
+    @property
+    def state_shape(self):
+        return (1, len(self.ifos), self.snapshotter.state_size)
+
+    @property
+    def inference_stride(self):
+        return int(self.sample_rate / self.inference_sampling_rate)
+
+    @property
+    def step_size(self):
+        return int(self.batch_size * self.inference_stride)
+
+    @property
+    def integration_size(self):
+        return int(self.integration_length * self.inference_sampling_rate)
+
+    @property
+    def window(self):
+        return np.ones((self.integration_size,)) / self.integration_size
+
+    @property
+    def times(self):
+        """
+        Returns the time values relative to event time
+        """
+        start = (
+            self.psd_length
+            + self.kernel_length
+            + (self.fduration / 2)
+            + self.padding
+        )
+        stop = self.kernel_length + (self.fduration / 2) + self.padding
+        return np.arange(-start, stop, 1 / self.sample_rate)
+
+    @property
+    def inference_times(self):
+        return self.times[:: self.inference_stride]
+
+    @property
+    def whitened_times(self):
+        start = (
+            self.step_size
+            - self.inference_stride
+            - int(self.sample_rate * self.fduration)
+        )
+        return self.times[start:]
+
+    def find_strain(self, time: float, shifts: Sequence[float]):
+        # find strain file corresponding to requested time
+        fname, t0, duration = get_strain_fname(self.strain_dir, time)
+        # find indices of data needed for inference
+        times = np.arange(t0, t0 + duration, 1 / self.sample_rate)
+        start, stop = get_indices(
```

Comment:
Won't `stop` be an index passed the end of the time array?

Reply:
Not sure what you mean

Reply:
Not sure what I meant by this either, to be honest. I think I misread something the first time I reviewed.

When is `time` relative to the coalescence point?